google-cloud-datastore/src/main/java/com/google/cloud/datastore/DatastoreImpl.java

```diff
@@ -132,8 +139,9 @@ void setPrevTransactionId(ByteString transactionId) {
 
     @Override
     public T call() throws DatastoreException {
-      transaction = datastore.newTransaction(options);
-      try {
+      try (com.google.cloud.datastore.telemetry.TraceUtil.Scope ignored =
+          transactionTraceContext.makeCurrent()) {
+        transaction = datastore.newTransaction(options);
```

Comment:
is there a difference in behavior if this is moved to inside the try block rather than before?

Reply:
Yes, `newTransaction` sends a `BEGIN_TRANSACTION` RPC to the backend and we want that to get traced.